pages/scenario_selection_page.py

```diff
@@ -309,67 +381,344 @@ def scenario_selection_page():
+            {
+                "lifetime_running_costs": (x["lifetime_running_costs"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "installation_costs_after_subsidy": (x["installation_costs_after_subsidy"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "loan_interest": (x["loan_interest"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "lifetime_maintenance_costs": (x["lifetime_maintenance_costs"] * x["number_of_properties"]).sum() / x["number_of_properties"].sum(),
+                "number_of_properties": x["number_of_properties"].sum(),
+            }
+        )
+    ).reset_index()
+    weighted_lifetime_costs["archetype_label"] = "Weighted average archetype"
+    weighted_lifetime_costs["life_span"] = (
+        lifetime_costs["life_span"]
+        .mean()
+        .round()
+        .astype(int)
+    )
+    lifetime_costs = pd.concat(
+        [lifetime_costs, weighted_lifetime_costs]
+    ).reset_index(drop=True)
+
+
+    lifetime_costs_selected_archetypes = lifetime_costs[
+        lifetime_costs["archetype_label"].isin(filter_archetypes)
     ]
 
-    st.dataframe(dummy_data, use_container_width=True)
-    st.dataframe(total_dummy_data, use_container_width=True)
+    df_comparing_costs = lifetime_costs_selected_archetypes.melt(
+        id_vars=["archetype_label", "technology", "life_span"],
+        value_vars=[
+            "lifetime_running_costs",
+            "installation_costs_after_subsidy",
+            "loan_interest",
+            "lifetime_maintenance_costs",
+        ],
+        var_name="cost_type",
+        value_name="cost",
+    )
+
+    cost_type_name_mapping = {
+        "lifetime_maintenance_costs": "Maintenance costs",
+        "loan_interest": "Loan interest",
+        "installation_costs_after_subsidy": "Installation costs (after subsidy)",
+        "lifetime_running_costs": "Running costs",
+
+    }
 
-    csv_annual = dummy_data.to_csv()
-    csv_total = total_dummy_data.to_csv()
+    df_comparing_costs["cost_type"] = df_comparing_costs["cost_type"].map(cost_type_name_mapping)
+
+    # Convert column to ordered categorical
+    df_comparing_costs["archetype_label"] = pd.Categorical(
+        df_comparing_costs["archetype_label"],
+        categories=mapped_archetype_options + ["Weighted average archetype"],
+        ordered=True
+    )
+    df_comparing_costs["cost_type"] = pd.Categorical(
+        df_comparing_costs["cost_type"],
+        categories=list(cost_type_name_mapping.values()),
+        ordered=True
+    )
+    df_comparing_costs = df_comparing_costs.sort_values(by=["archetype_label", "cost"])
+
+    cost_component_order = {
+        "Loan interest":0,
+        "Maintenance costs":1,
+        "Installation costs (after subsidy)":2,
+        "Running costs":3,
+    }
+    df_comparing_costs["cost_type_order"] = df_comparing_costs["cost_type"].map(cost_component_order)
+
```

Comment:
As suggested in the feedback slide (sprint 4 slide deck), I think it is worth separating out the total of all cost components. We can also have this displayed in the altair chart tooltips box

```suggestion
    df_comparing_costs["total_cost"] = df_comparing_costs.groupby(
        ["technology", "archetype_label"]
    )["cost"].transform("sum")
```